app/desktop/src/main/kotlin/AniDesktop.kt

```diff
@@ -445,13 +424,33 @@ object AniDesktop {
     }
 }
 
+@OptIn(InternalComposeUiApi::class)
 @Composable
 private fun FrameWindowScope.MainWindowContent(
     aniNavigator: AniNavigator,
 ) {
     val settingsRepository = KoinPlatform.getKoin().get<SettingsRepository>()
     AniApp {
-        window.setTitleBar(AniThemeDefaults.navigationContainerColor, isSystemInDarkTheme())
+        val themeSettings = LocalThemeSettings.current
+        val titleBarThemeController = LocalTitleBarThemeController.current
+        val systemTheme = LocalSystemTheme.current
+        val navContainerColor = AniThemeDefaults.navigationContainerColor
+
+        val isTitleBarDark = remember(themeSettings, systemTheme) {
+            when (themeSettings.darkMode) {
+                DarkMode.AUTO -> systemTheme == SystemTheme.Dark
+                DarkMode.LIGHT -> false
+                DarkMode.DARK -> true
+            }
+        }
+        val owner = remember { Any() }
+        DisposableEffect(isTitleBarDark, titleBarThemeController, owner) {
+            window.setTitleBar(navContainerColor, isTitleBarDark)
+            titleBarThemeController?.requestTheme(owner = owner, isDark = isTitleBarDark)
+            onDispose {
+                titleBarThemeController?.removeTheme(owner = owner)
+            }
+        }
```

Comment:
我觉得可以把 `DarkCaptionButtonAppearance` 改成 `OverrideCaptionButtonAppearance(isDark)` 之类的, 用参数传递期望的颜色, 而不是固定 dark. 这样也更能支持以后可能需要浅色的页面. 

然后这里可以改成 (separate concern):

```suggestion
        DisposableEffect(isTitleBarDark, navContainerColor) {
            window.setTitleBar(navContainerColor, isTitleBarDark)
            onDispoase {}
        }
        OverrideCaptionButtonAppearance(isTitleBarDark)
```